src/darwin/Framework/CHIP/MTRAttributeValueWaiter.mm

```diff
@@ -70,7 +78,13 @@ - (void)dealloc
 
 - (void)cancel
 {
-    [self.device _attributeWaitCanceled:self];
+    [self.device _forgetAttributeWaiter:self];
+    [self _cancelWithoutRemovingFromDevice];
+}
+
+- (void)_cancelWithoutRemovingFromDevice
```

Comment:
Nit: Could call this _notifyCancellation since that's really all it does.

Reply:
Good idea, done.